src/test/java/com/bbtutorials/users/dao/WorkoutDaoTests.java

```diff
@@ -0,0 +1,142 @@
+                        new AttributeDefinition().withAttributeName("Id").withAttributeType(ScalarAttributeType.S))
+                .withProvisionedThroughput(new ProvisionedThroughput(5L, 5L))
+        );
+
+        workoutDao = new WorkoutDao(mapper);
+    }
+
+    public Workout createWorkout() {
+        Workout testWorkout = new Workout();
+
+        testWorkout.setUserEmail(testEmail);
+        testWorkout.setName(testName);
+        
+        Set<String> testExer = new HashSet<String>();
+        Set<Integer> testDays = new HashSet<Integer>();
+        
+        testExer.add("testexercise1");
+        testExer.add("testexercise2");
+        testWorkout.setExercises(testExer);
+        
+        testDays.add(1);
+        testDays.add(2);
+        testWorkout.setDaysOfWeek(testDays);
+
+        return testWorkout;
+    }
+
+
+    @Test
+    public void testPutGetRemoveWorkout() {
+        workoutDao.putWorkout(createWorkout());
+
+        List<Workout> workouts = workoutDao.getWorkouts(testEmail);
+        
+        Set<Integer> checkDays = workouts.get(0).getDaysOfWeek();
+        Set<String> checkExer = workouts.get(0).getExercises();
+        assertEquals(1, workouts.size());
+        assertEquals(testName, workouts.get(0).getName());
+
+        assertEquals(testEmail, workouts.get(0).getUserEmail());
+        assertEquals(2, checkDays.size());
+        assertEquals(2, checkExer.size());
+        assertTrue(checkDays.contains(2));
+        assertTrue(checkDays.contains(1));
+        assertTrue(checkExer.contains("testexercise1"));
+        assertTrue(checkExer.contains("testexercise2"));
+        
+        workoutDao.removeWorkout(workouts.get(0));
+
+        workouts = workoutDao.getWorkouts(testEmail);
+
+        assertTrue(workouts.isEmpty());
+    }
+
+    @Test
+    public void testMultipleWorkouts() {
+        workoutDao.putWorkout(createWorkout());
+        Workout second = createWorkout();
+        //String newName = "newName";
+        //second.setId(newName);
+
+        workoutDao.putWorkout(second);
+
+        List<Workout> workouts = workoutDao.getWorkouts(testEmail);
+        
+        //assertEquals(newName, workouts.get(1).getName());
+
+        assertEquals(2, workouts.size());
+        
+        for (Workout toRemove : workouts) {
+            workoutDao.removeWorkout(toRemove);
+        }
+    }
+
+    @Test
+    public void testUpdateWorkout() {
+        workoutDao.putWorkout(createWorkout());
+
+        List<Workout> workouts = workoutDao.getWorkouts(testEmail);
+        Set<Integer> testDays2 = new HashSet<Integer>();
+        
+        testDays2.add(3);
+        testDays2.add(4);
+        
+        Workout updatedWorkout = workouts.get(0);
+        updatedWorkout.setDaysOfWeek(testDays2);
+
+        workoutDao.updateWorkout(updatedWorkout);
+
+        workouts = workoutDao.getWorkouts(testEmail);
+
+        Set<Integer> checkDays = workouts.get(0).getDaysOfWeek();
+        assertTrue(checkDays.contains(3));
```

Comment:
Please use assertEquals() for two hash sets here

Reply:
I think assertTrue works fine. Why is assertEquals better?

Reply:
In this case, all you are doing is checking if the hash set contains the two elements 3 and 4, but what if the update API implementation changed such that the existing set and new set were concatenated, etc.? The test would still pass in that case but we wouldn't want it to pass since there would be extra elements in the set we didn't check for